### DISCUSSION TITLE
Expose and use new "Valve" entity type as another device type within Home Assistant's ZHA integration?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#1314] [ZHA development]
Home Assistant 2024.1 release added `valve` as a brand new entity type, would it be a good idea if was exposed under ZHA integration?

https://www.home-assistant.io/blog/2024/01/03/release-20241/#new-entity-type-valve

This device type in Home Assistant is meant to provide a more clearer defined interface to control valves such as water, gas, or air valves:

https://www.home-assistant.io/integrations/valve

Initially this new Valves so far support the following device classes:

- _**None**: Generic valve. This is the default and doesn’t need to be set._
- _**water**: Valve that controls the flow of water through a system._
- _**gas**: Valve that controls the flow of gas through a system._

If supported in ZHA then users can at least manually modify any `switch` device type that controls a valve device type and device class so that it is presented as a valve by changing it to `valve` in the customize section of ZHA UI or the device config in `configuration.yaml`:

https://www.home-assistant.io/docs/configuration/customizing-devices/

https://www.home-assistant.io/integrations/zha#modifying-the-device-type

Not actually sure if any existing Zigbee devices specifically expose many valve specific attributes ZCL (Zigbee Cluster Library) but there are valve-specific products, especially related to water and gas controllers, such as example these water and gas shutoff products:

* ShinaSystem GCM-300Z (SiHAS gas valve) -> https://www.zigbee2mqtt.io/devices/GCM-300Z.html
* RTX ZVG1 (Zigbee smart water valve) https://www.zigbee2mqtt.io/devices/ZVG1.html
* QOTO QT-05M (Solar powered garden watering timer) -> https://www.zigbee2mqtt.io/supported-devices/#e=valve_state_auto_shutdown

There are many generic valve shut off products that that simply expose either state and position or just on and off via a switch:

* Sinopé VA4220ZB (Sedna smart water valve) -> https://www.zigbee2mqtt.io/devices/VA4220ZB.html
* Sinopé VA4201WZ (Zigbee smart water valve) -> https://www.zigbee2mqtt.io/devices/VA4201WZ.html
* GiEX QT06_2 (Water irrigation valve) ->  https://www.zigbee2mqtt.io/devices/QT06_2.html
* GiEX QT06_1 (Water irrigation valve) https://www.zigbee2mqtt.io/devices/QT06_1.html
* IKEA E1842 (KNYCKLAN receiver electronic water valve shut-off) -> https://www.zigbee2mqtt.io/devices/E1842.html
* IOTPerfect PF-PM02D-TYZ (Smart water/gas valve) -> https://www.zigbee2mqtt.io/devices/PF-PM02D-TYZ.html
* FrankEver FK_V02 (Zigbee smart water valve) https://www.zigbee2mqtt.io/devices/FK_V02.html
* Waxman 8850100 (leakSMART automatic water shut-off valve 2.0) -> https://www.zigbee2mqtt.io/devices/8850100.html
* TuYa TS0111_valve (Smart water/gas valve) -> https://www.zigbee2mqtt.io/devices/TS0111_valve.html
* TuYa ZN231392 (Smart water/gas valve) -> https://www.zigbee2mqtt.io/devices/ZN231392.html

PS: In the future this may also be extended to support additinal device classes that valves are also commonly used for, such example:

* heat: for other heating valves not directly related to climate (on means hot, off means normal), like for example to/from a water heater or pool.
* cold: for other cooling valves not directly related to climate (on means hot, off means normal)

**Comment by claudegel** [comment]
I already added Sinopé valves in zha-devices-handler but until now it is in switch.py. I already created the valve.py to support all Sinopé valves as valve entities but I'm waiting for ZHA to implement valve entities.
